tests/tools/test_search.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+import pathlib
+import random
+import uuid
+from typing import Iterator, Literal
+from unittest.mock import patch
+
+import pytest
+
+# try importing lancedb before importing `dlt_mcp` modules using it
+lancedb = pytest.importorskip(
+    "lancedb", reason="`lancedb` is required to run search tools tests."
+)
+
+from dlt_mcp._utilities import ingestion  # noqa: E402
+from dlt_mcp._tools import search  # noqa: E402
+
+
+@pytest.fixture(scope="module")
+def module_tmp_path(tmp_path_factory) -> pathlib.Path:
+    return tmp_path_factory.mktemp("dlt-mcp_test_search")
+
+
+# this fixture must `yield` to ensure the `with patch` context is
+# preserved in the test function
+@pytest.fixture(scope="module")
+def populated_local_db(module_tmp_path: pathlib.Path) -> Iterator[lancedb.Table]:
+    # the two tables need to be populated from the same fixture to avoid
+    # conflicts with the utility `_maybe_ingest_docs_and_code`.
+    with patch("pathlib.Path.home", return_value=module_tmp_path):
+        db_con = ingestion.db_con(ingestion.DLT_VERSION)
+        # ingest docs
+        docs_chunks = [
+            dict(
+                id=str(uuid.uuid4()),
+                text=f"This is a test chunk {i}",
+                token_count=100,
+                file_path="test.md",
+                start_index=i * 100,
+                end_index=(i + 1) * 100,
+                chunk_index=i,
+                vector=[
+                    random.random()
+                    for _ in range(ingestion._lancedb_embedding_model.ndims())
+                ],
+            )
+            for i in range(10)
+        ]
+        ingestion.page_chunks_table(db_con, docs_chunks)
+
+        # ingest code
+        code_chunks = [
+            dict(
+                id=str(uuid.uuid4()),
+                text=f"This is a test chunk {i}",
+                token_count=100,
+                # `file_path` has to start with `dlt/` to be picked up
+                file_path=f"dlt/{i % 2}/test.md",
+                start_index=i * 100,
+                end_index=(i + 1) * 100,
+                chunk_index=i,
+                vector=[
+                    random.random()
+                    for _ in range(ingestion._lancedb_embedding_model.ndims())
+                ],
+            )
+            for i in range(10)
+        ]
+        ingestion.code_chunks_table(db_con, code_chunks)
+
+        yield db_con
+
+
+@pytest.mark.parametrize("mode", ["full_text", "hybrid", "vector"])
+def test_docs_search(
+    populated_local_db: lancedb.LanceDBConnection,
+    mode: Literal["full_text", "hybrid", "vector"],
+):
+    if mode == "full_text":
```

Comment:
this could just be a dictionary `dict[str, (scoring_field, lower_is_better)` or a match statement in a private function it would help improve the readability flow of the test